src/app/controllers/reviewGraph.js

```diff
@@ -0,0 +1,1198 @@
+                $scope.clickCriticalPath = function() {
+                    $scope.manageCriticalPath = !$scope.manageCriticalPath
+                    //$scope.hideSidebarWhenCriticalPath = !$scope.hideSidebarWhenCriticalPath
+                }
+
+                var resetGraphAttributes = function() {
+                    // reset attributes of each node (opacity, opacityFilter)
+                    nodes.forEach(n => {
+                        var np = Object.getPrototypeOf(n);
+                        (np.opacity = np.originalOpacity) && (np.opacityFilter = false);
+                    });
+                    updateNodeIconOpacity(d3v5.select("#nodeGroup_" + graph.id).selectAll("use"));
+                    // reset attributes of each link (opacity, opacityFilter)
+                    links.forEach(l => {
+                        var lp = Object.getPrototypeOf(l);
+                        (lp.opacity = lp.originalOpacity) && (lp.opacityFilter = false);
+                    });
+                    updateLinkIconOpacity(d3v5.select("#linkGroup_" + graph.id).selectAll("polyline"));
+                    // reset activation flag of each node
+                    nodes.forEach(d=> Object.getPrototypeOf(d).enabledNode = true)
+                    // reset the sidebar attributes
+                    var sidebarIcons = d3v5.selectAll("img")
+                    sidebarIcons.nodes().filter(d => d.id.includes('sideButton'))
+                    // reset the buttons opacity
+                    sidebarIcons.attr("style", d => "opacity:" + "1")
+                    // reset the state of the sidebar icons
+                    sidebarIcons.nodes().forEach(d => d.iconState = false)
+                }
+
+                // manage the criticalPath activation
+                $scope.$watch("manageCriticalPath", function(newVal, oldVal) {
+                    resetGraphAttributes()
+                    var mainNode = (Object.getPrototypeOf(nodes.filter(n=> n.id == graph.id)[0]))
+                    let criticalPath = findCriticalPath(mainNode)
+                    var graphNodesToManage = nodes.filter(n => !criticalPath.includes(Object.getPrototypeOf(n)));
+                    if (newVal != oldVal) {
+                        if (newVal) {
+                            graphNodesToManage.forEach(d=> Object.getPrototypeOf(d).enabledNode = false)
+                            var criticalpath = true
+                            handleNodeActivation(graphNodesToManage, criticalpath);
+                            handleLinkActivation(graphNodesToManage, criticalpath);
+                            $scope.criticalPathButtonText = "Click to show the whole reviewGraph"
+                        }
+                        else {
+                            $scope.criticalPathButtonText = "Click to show the criticalPath"
+                        }
+                    }
+                });
+
+            };
+        });
+    }
+);
```

Comment:
In general, this is much cleaner and the comments help to understand what's happening. 
As a rule of thumb, it's best to avoid files longer than 500 lines as this means you're trying to do too much in one place (and if you want to change something, you have to find the right line and make sure that you're not breaking something else in the file). For now, maybe add a TODO line at the top, documenting how you would split this current implementation into 2 or 3 smaller files with their own separate responsabilities and how they could communicate with each other to achieve the same functionality.

Reply:
Ok, added a TODO at the top.